Movies_MVVMTests/DataServiceTests.swift

```diff
@@ -0,0 +1,89 @@
+// DataServiceTests.swift
+// Copyright © Aleksandr Dorofeev. All rights reserved.
+
+import XCTest
+
+@testable import Movies_MVVM
+
+/// Data service tests
+final class DataServiceTests: XCTestCase {
+    // MARK: - Private Constants
+
+    private enum Constants {
+        static let emptyString = ""
+        static let mockCategory = "mockCategory"
+        static let fooData = "Foo"
+        static let barData = "Bar"
+        static let mockMovieId = 2
+        static let mockMovieDetailId = 1
+        static let mockGenres: [MovieDetail.Genres] = []
+    }
+
+    // MARK: - Private properties
+
+    private var mockMovies: [Movie] = [
+        Movie(
+            id: Constants.mockMovieId,
+            posterPath: Constants.barData,
+            title: Constants.barData,
+            voteAverage: 1
+        )
+    ]
+    private var mockMovieDetail = MovieDetail(
+        posterPath: Constants.fooData,
+        title: Constants.fooData,
+        voteAverage: 0,
+        releaseDate: Constants.fooData,
+        genres: Constants.mockGenres,
+        overview: Constants.fooData,
+        id: Constants.mockMovieDetailId
+    )
+    private var dataService: DataServiceProtocol?
+
+    // MARK: - Public methods
+
+    override func setUp() {
+        dataService = DataService()
+    }
+
+    override func tearDown() {
+        dataService = nil
+    }
+
+    func testWriteMovies() {
+        XCTAssertNotNil(dataService?.writeMovieData(movies: mockMovies, category: Constants.mockCategory))
```

Comment:
```suggestion
        XCTAssertNotNil(dataService?.writeMovieData(movies: mockMovies, category: Constants.mockCategory))
```
Этот метод ничего не возвращает - тест будет пройден всегда

Reply:
Этот тест проверяет вызывается ли метод, при тесте реального объекта я могу только это в сервисе проверить, я добавил мок сервиса, но ожидаемо на пару процентов упал процент покрытия,  потому что я тестирую абстракцию а не сам сервис